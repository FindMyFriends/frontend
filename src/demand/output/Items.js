import React from 'react';
import PropTypes from 'prop-types';
import { requestedConfirm } from './../../ui/actions';
import { retract } from './../../demand/endpoints';
import { ActionIcon } from './../../components/menu/Icon';

const handleRetract = (history, id) => (dispatch) => {
  dispatch(requestedConfirm(
    'Are you sure, you want to retract demand?',
    () => dispatch(retract(id, history)),
  ));
};

export const ActionItems = ({ history, id, dispatch }) => (
  <React.Fragment>
    <ActionIcon className="material-icons" onClick={() => history.push(`/demands/${id}/edit`)}>edit</ActionIcon>
    <ActionIcon className="material-icons" style={{ color: '#F44336' }} onClick={() => dispatch(handleRetract(history, id))}>delete</ActionIcon>
  </React.Fragment>
);

ActionItems.propTypes = {
  history: PropTypes.object.isRequired,
  id: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
};
