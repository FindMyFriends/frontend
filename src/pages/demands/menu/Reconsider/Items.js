import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ActionIcon } from '../../../../components/menu/Icon';

export const ActionItems = ({ id }) => (
  <React.Fragment>
    <Link to={`/demands/${id}`}>
      <ActionIcon className="material-icons" title="Demand">visibility</ActionIcon>
    </Link>
  </React.Fragment>
);

ActionItems.propTypes = {
  id: PropTypes.string.isRequired,
};
