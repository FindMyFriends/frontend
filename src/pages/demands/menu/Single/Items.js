import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { requestedConfirm } from '../../../../ui/actions';
import { retract } from '../../../../demand/endpoints';
import { ActionIcon, DangerIcon } from '../../../../components/menu/Icon';

const handleRetract = (history, id) => (dispatch) => {
  dispatch(requestedConfirm(
    'Are you sure, you want to retract demand?',
    () => dispatch(retract(id, history)),
  ));
};

export const ActionItems = ({ history, id, dispatch }) => (
  <React.Fragment>
    <Link to={`/demands/${id}/edit`}>
      <ActionIcon className="material-icons">edit</ActionIcon>
    </Link>
    <DangerIcon className="material-icons" onClick={() => dispatch(handleRetract(history, id))}>delete</DangerIcon>
  </React.Fragment>
);

ActionItems.propTypes = {
  history: PropTypes.object.isRequired,
  id: PropTypes.string.isRequired,
  dispatch: PropTypes.func.isRequired,
};
