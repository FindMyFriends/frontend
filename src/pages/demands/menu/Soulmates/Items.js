import React from 'react';
import PropTypes from 'prop-types';
import * as R from 'ramda';
import { ActionIcon, InActionIcon } from '../../../../components/menu/Icon';

const isSeeking = request => request.status === 'pending' || request.status === 'processing';

export const RefreshButton = ({ requests, onRefresh }) => {
  const request = requests[0];
  if (R.isEmpty(request)) {
    return null;
  }
  if (isSeeking(request)) {
    return (
      <InActionIcon title="In progress" className="material-icons">refresh</InActionIcon>
    );
  } else if (request.is_refreshable) {
    return (
      <ActionIcon className="material-icons" onClick={onRefresh}>refresh</ActionIcon>
    );
  }
  return (
    <InActionIcon title="Not available" className="material-icons">refresh</InActionIcon>
  );
};

export const ActionItems = ({ requests, onRefresh }) => (
  <React.Fragment>
    <RefreshButton requests={requests} onRefresh={onRefresh} />
  </React.Fragment>
);


RefreshButton.propTypes = {
  requests: PropTypes.array.isRequired,
  onRefresh: PropTypes.func.isRequired,
};

ActionItems.propTypes = {
  requests: PropTypes.array.isRequired,
  onRefresh: PropTypes.func.isRequired,
};
