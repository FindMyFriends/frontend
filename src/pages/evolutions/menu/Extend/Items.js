import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import VisibilityIcon from 'material-ui/svg-icons/action/visibility';

export const ActionItems = ({ id }) => (
  <Link to={`/evolutions/${id}`}>
    <VisibilityIcon />
  </Link>
);

ActionItems.propTypes = {
  id: PropTypes.string.isRequired,
};
