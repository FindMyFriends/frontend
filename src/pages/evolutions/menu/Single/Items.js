import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import DeleteIcon from 'material-ui/svg-icons/action/delete';
import EditIcon from 'material-ui/svg-icons/image/edit';
import { red500, white } from 'material-ui/styles/colors';

export const ActionItems = ({ id }) => (
  <React.Fragment>
    <Link to={`/evolutions/${id}/change`}>
      <EditIcon style={{ padding: 10 }} color={white} />
    </Link>
    <DeleteIcon
      color={red500}
      style={{ padding: 10, cursor: 'pointer' }}
      onClick={() => {}}
    />
  </React.Fragment>
);

ActionItems.propTypes = {
  id: PropTypes.string.isRequired,
};
