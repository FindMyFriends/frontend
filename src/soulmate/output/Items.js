import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledIcon = styled.i`
  cursor: pointer;
  color: white;
  padding: 10px;
`;

const InActionIcon = styled.i`
  cursor: not-allowed;
  color: #757575;
  padding: 10px;
`;

const isSeeking = request => request && (request.status === 'pending' || request.status === 'processing');

export const RefreshButton = ({ requests, onRefresh }) => {
  const request = requests[0];
  if (request) {
    if (isSeeking(request)) {
      return (
        <InActionIcon title="In progress" className="material-icons">refresh</InActionIcon>
      );
    }
    return (
      <StyledIcon className="material-icons" onClick={onRefresh}>refresh</StyledIcon>
    );
  }
  return null;
};

export const ActionItems = ({ requests, onRefresh }) => (
  <React.Fragment>
    <RefreshButton requests={requests} onRefresh={onRefresh} />
  </React.Fragment>
);


RefreshButton.propTypes = {
  requests: PropTypes.array.isRequired,
  onRefresh: PropTypes.func.isRequired,
};

ActionItems.propTypes = {
  requests: PropTypes.array.isRequired,
  onRefresh: PropTypes.func.isRequired,
};
