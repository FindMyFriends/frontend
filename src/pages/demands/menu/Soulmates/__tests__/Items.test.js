import React from 'react';
import renderer from 'react-test-renderer';
import 'raf/polyfill';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { RefreshButton } from '../Items';

it('ready refresh indicator for available refresh', () => {
  const component = (
    <MuiThemeProvider>
      <RefreshButton
        requests={[{ is_refreshable: true }]}
        onRefresh={() => {}}
      />
    </MuiThemeProvider>
  );
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('not available indicator for done task', () => {
  const component = (
    <MuiThemeProvider>
      <RefreshButton
        requests={[{ is_refreshable: false, is_seeking: false }]}
        onRefresh={() => {}}
      />
    </MuiThemeProvider>
  );
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('blocked indicator for pending task', () => {
  const component = (
    <MuiThemeProvider>
      <RefreshButton
        requests={[{ is_refreshable: false, is_seeking: true }]}
        onRefresh={() => {}}
      />
    </MuiThemeProvider>
  );
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('ready refresh indicator for available refresh', () => {
  const component = (
    <MuiThemeProvider>
      <RefreshButton
        requests={[{ is_refreshable: true, is_seeking: false }]}
        onRefresh={() => {}}
      />
    </MuiThemeProvider>
  );
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('nothing for empty request', () => {
  const component = (
    <MuiThemeProvider>
      <RefreshButton
        requests={[{ }]}
        onRefresh={() => {}}
      />
    </MuiThemeProvider>
  );
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('first block for seeking', () => {
  const component = (
    <MuiThemeProvider>
      <RefreshButton
        requests={[{ is_refreshable: true, is_seeking: true }]}
        onRefresh={() => {}}
      />
    </MuiThemeProvider>
  );
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
