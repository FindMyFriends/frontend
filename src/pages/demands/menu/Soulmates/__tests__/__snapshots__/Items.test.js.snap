// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blocked indicator for pending task 1`] = `
<i
  className="material-icons sc-bxivhb xBree"
  title="In progress"
>
  refresh
</i>
`;

exports[`first block for seeking 1`] = `
<i
  className="material-icons sc-bxivhb xBree"
  title="In progress"
>
  refresh
</i>
`;

exports[`not available indicator for done task 1`] = `
<i
  className="material-icons sc-bxivhb xBree"
  title="Not available"
>
  refresh
</i>
`;

exports[`nothing for empty request 1`] = `null`;

exports[`ready refresh indicator for available refresh 1`] = `
<i
  className="material-icons sc-bdVaJa fcJCzI"
  onClick={[Function]}
>
  refresh
</i>
`;

exports[`ready refresh indicator for available refresh 2`] = `
<i
  className="material-icons sc-bdVaJa fcJCzI"
  onClick={[Function]}
>
  refresh
</i>
`;
